src/methods/take.ts

```diff
@@ -1,9 +1,12 @@
 import { filter } from '..';
-import { TransformTyped } from '../types';
+import { PureStream } from '../PureStream';
 
-export function take<T>(skip: number, count: number): TransformTyped<T, T>
-export function take<T>(count: number): TransformTyped<T, T>
-export function take<T>(skip: number, count?: number): TransformTyped<T, T> {
+/**
+ * Only take a certain number of items from a stream
+ */
+export function take<T>(skip: number, count: number): PureStream<T, T>
+export function take<T>(count: number): PureStream<T, T>
+export function take<T>(skip: number, count?: number): PureStream<T, T> {
```

Comment:
This might be more clear if written to match the api and wording of [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice).

Reply:
Sure! In that case a separate `slice` method would probably make this the clearest.

Reply:
The issue I had with this signature is that the first variable is either 'skip' or 'count', which changes the behaviour - it would be more consistent if the first argument had the same meaning over all of the polys

Reply:
Yeah I agree.
Splitting it into `take(count: number)` and `slice(begin: number, end?: number)` would be less ambiguous.